app/scripts/lib/tx-verification/tx-verification-middleware.ts

```diff
@@ -49,55 +60,55 @@ export function createTxVerificationMiddleware(
 
     // the tx object is the first element
     const params = req.params[0];
-
     const chainId =
       typeof params.chainId === 'string'
-        ? (params.chainId.toLowerCase() as `0x${string}`)
+        ? (params.chainId.toLowerCase() as Hex)
         : networkController.state.providerConfig.chainId;
 
-    // skip verification if bridge is not deployed on the specified chain.
-    // skip verification to address is not the bridge contract
-    if (
-      !Object.keys(FIRST_PARTY_CONTRACT_NAMES['MetaMask Bridge']).includes(
-        chainId,
-      ) ||
-      params.to.toLowerCase() !==
-        FIRST_PARTY_CONTRACT_NAMES['MetaMask Bridge'][chainId].toLowerCase()
-    ) {
+    const r = addrToExpMap[params.to.toLowerCase()];
+    // if undefined then no address matched
+    if (!r) {
+      return next();
+    }
+    const { experienceType, chainId: experienceChainId } = r;
+    // skip if chainId is different
+    if (experienceChainId !== chainId) {
+      return next();
+    }
+    // skip if experience is not one we want to verify against
+    if (!EXPERIENCES_TO_VERIFY.includes(experienceType as EXPERIENCES_TYPE)) {
       return next();
     }
 
-    const paramsToVerify = {
-      to: hashMessage(params.to.toLowerCase()),
-      from: hashMessage(params.from.toLowerCase()),
-      data: hashMessage(
-        params.data.toLowerCase().slice(0, params.data.length - SIG_LEN),
-      ),
-      value: hashMessage(params.value.toLowerCase()),
-    };
-    const hashedParams = hashMessage(JSON.stringify(paramsToVerify));
-
-    // signature is 130 chars in length at the end
-    const signature = `0x${params.data.slice(-SIG_LEN)}`;
-    const addressToVerify = verifyMessage(hashedParams, signature);
-
-    if (addressToVerify.toLowerCase() !== TRUSTED_BRIDGE_SIGNER.toLowerCase()) {
-      return end(
-        rpcErrors.invalidParams('Invalid bridge transaction signature.'),
-      );
+    const signature = `0x${params.data.slice(-TX_SIG_LEN)}`;
+    const addressToVerify = verifyMessage(hashedParams(params), signature);
+    if (addressToVerify !== trustedSigners[experienceType]) {
```

Comment:
Don't mean to complicate things, but is it likely we'll have different signers per chain for the same experience?

Reply:
It's possible, but it's very unlikely. Mainly because there's no point for a single service to handle individual keys across networks and instead it's likely that they'll have a single signer that signs across all networks.

In the event if we do I thought turning `TRUSTED_SIGNERS: Partial<Record<EXPERIENCES_TYPE, Hex>>` into 
`TRUSTED_SIGNERS: Partial<Record<EXPERIENCES_TYPE, Hex[]>>` would be relatively trivial.